dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java

```diff
@@ -255,6 +255,15 @@ private String resultProcess(ResultSet resultSet) throws Exception {
                 resultJSONArray.add(mapOfColValues);
                 rowCount++;
             }
+            // result is null,output table metadata
+            if (resultJSONArray.isEmpty()) {
+                ObjectNode emptyOfColValues = JSONUtils.createObjectNode();
+                for (int i = 1; i <= num; i++) {
+                    emptyOfColValues.set(md.getColumnLabel(i), JSONUtils.toJsonNode(""));
+                }
+                resultJSONArray.add(emptyOfColValues);
```

Comment:
If put the empty row, the display rows is 1, but it's 0 in fact.

Reply:
If the data is empty, I want to send Excel with header
![image](https://user-images.githubusercontent.com/35831367/191250803-cc7373bc-de01-4dc0-9c5b-35fc14690dec.png)

Reply:
Yes, I get your point, but what I mean is that the log will show the incorrect empty result. It's better to set empty value before sendAttachment. see https://github.com/apache/dolphinscheduler/pull/12059/files#diff-f54c9e235dbb7e6ab2bd181aaad09f19330e680b8573840d64386401abb47203R269-R273

BTW, I think you can split this code as a function, like `generateEmptyRow`, WDYT?

Reply:
yes, I got it